node/src/components/consensus/highway_core/state.rs

```diff
@@ -634,7 +698,7 @@ impl<C: Context> State<C> {
                 Observation::Faulty
             }
         };
-        self.panorama[wunit.creator] = new_obs;
+        self.panoramas.update_panorama(wunit.creator, new_obs);
```

Comment:
Maybe the whole `update_panorama` logic should move to `Panoramas` now? Part of it will also be needed by `Panoramas::endorsed`.

Reply:
I wanted to but it calls `find_in_swimlane` and `wire_unit` which are methods on `State`. I didn't want to change too much.